marlowe-dashboard-client/src/Contract/View.purs

```diff
@@ -1083,3 +1080,91 @@ getParty (S.IDeposit _ p _ _) = Just p
 getParty (S.IChoice (ChoiceId _ p) _) = Just p
 
 getParty _ = Nothing
+
+-------------------------------------------------------------------------------
+-- UI Layout helpers
+-------------------------------------------------------------------------------
+withBaseCss ::
```

Comment:
What is the idea behind `withBaseCss`? Why not just have `div [ classNames ["grid",...]]`

Reply:
This idea was arrived at by refactoring and extracting patterns a few times:

Starting point:

```purescript
titleBar1 = header [ classNames [ "p-4", "border", "border-gray", "flex", "items-center", "justify-between" ] ]
  [ text "I'm title #1"
  , icon ...
  ]

...

titleBar2 = header [ classNames [ "p-4", "border", "border-gray" ] ]
  [ text "I'm title #2"
  ]

...

titleBar3 = header [ classNames [ "p-4", "border", "border-gray" ] ]
  [ text "I'm title #3"
  ]

...

body1 = section [ classNames [ "grid", "p-4", "bg-white" ] ]
  [ text "I'm body #1"
  , icon ...
  ]

...

body2 = section [ classNames [ "grid", "p-4", "bg-gray" ] ]
  [ text "I'm body #2"
  , icon ...
  ]
```

Pattern: multiple fragments of HTML using the same styles
Goal: Collect related styles into reusable helper
Refactor: extract helpers to create components with same "container" styles:

```purescript
titleBar css = header [ classNames [ "p-4", "border", "border-gray" ] <> css ]

body css = section [ classNames [ "grid", "p-4" ] <> css ]

titleBar1 = titleBar [ "flex", "items-center", "justify-between" ]
  [ text "I'm title #1"
  , icon ...
  ]

...

titleBar2 = titleBar [] [ text "I'm title #2" ]

...

titleBar3 = titleBar [] [ text "I'm title #3" ]

...

body1 = body  [ "bg-white" ]
  [ text "I'm body #1"
  , icon ...
  ]

...

body2 = body [ "bg-gray" ]
  [ text "I'm body #2"
  , icon ...
  ]
```

Pattern: Multiple helpers which differ only in the element rendered, and the base set of styles
Refactor: Extract pattern into single helper

```purescript
withBaseCss el baseCss css = el [ classNames $ baseCss <> css ]

titleBar = header `withBaseCss` [ "p-4", "border", "border-gray" ]

body = section `withBaseCss` [ "grid", "p-4" ]

titleBar1 = titleBar [ "flex", "items-center", "justify-between" ]
  [ text "I'm title #1"
  , icon ...
  ]

...

titleBar2 = titleBar [] [ text "I'm title #2" ]

...

titleBar3 = titleBar [] [ text "I'm title #3" ]

...

body1 = body  [ "bg-white" ]
  [ text "I'm body #1"
  , icon ...
  ]

...

body2 = body [ "bg-gray" ]
  [ text "I'm body #2"
  , icon ...
  ]
```

That's the idea, anyway. I think perhaps 80% of the value is in the first refactor, and the second one isn't as helpful. It might even be detrimental because of the additional layer of indirection (and I think in this example, it actually increases the overall amount of code)